processing/src/main/java/org/apache/druid/segment/data/VSizeLongSerde.java

```diff
@@ -413,9 +412,81 @@ public void close() throws IOException
     }
   }
 
+  /**
+   * Unpack bitpacked long values from an underlying contiguous memory block
+   */
   public interface LongDeserializer
   {
+    /**
+     * Unpack long value at the specified row index
+     */
     long get(int index);
+
+    /**
+     * Unpack a contiguous vector of long values at the specified start index of length and adjust them by the supplied
+     * delta base value.
+     */
+    default void getDelta(long[] out, int outPosition, int startIndex, int length, long base)
```

Comment:
Are the default implementations ever used? If not, we could remove them.

Reply:
i think the default getDelta methods could be removed, and the default getTable methods could probably throw an unsupported operation exception for sizes that don't support table encoding (or we could push that to the implementations so no defaults)

Reply:
I was wrong, the default contiguous `getDelta` methods are not used, but `int[] indexes` is used by all un-aligned decoders (1,2,4,12,20,24,40,48,56)